src/rtos/task_notify_when_deleting.c

```diff
@@ -41,15 +44,39 @@ void task_notify_when_deleting(task_t target_task, task_t task_to_notify,
   task_to_notify = (task_to_notify == NULL) ? pxCurrentTCB : task_to_notify;
   target_task = (target_task == NULL) ? pxCurrentTCB : target_task;
 
-  // It doesn't make sense for a task to notify itself, and make sure that neither task is NULL (implying that scheduler hasn't started yet)
+  // It doesn't make sense for a task to notify itself, and make sure that
+  // neither task is NULL (implying that scheduler hasn't started yet)
   if (task_to_notify == target_task || !task_to_notify || !target_task) {
     return;
   }
 
-  linked_list_s_t* target_ll = pvTaskGetThreadLocalStoragePointer(target_task, TLSP_IDX);
+  // task_to_notify maintains a list of the tasks whose deletion it cares about.
+  // This will allow us to unsubscribe from notification if/when task_to_notify
+  // is deleted
+  linked_list_s_t* subscriptions_ll = pvTaskGetThreadLocalStoragePointer(task_to_notify, SUBSCRIPTIONS_TLSP_IDX);

```

Comment:
So subscriptions_ll is the linked list representing what tasks will notify `task_to_notify` when they die. What is the payload?

Reply:
payload is just the task handle

Reply:
So subscriptions is a list of tasks `task_to_notify` has requested to be notified of when any one of them have died?

Reply:
yes. that way when `task_to_notify` dies, it can iterate through this list and remove itself from being notified, clearing up any issues of UB that would result from it dying before other tasks try to notify it

Reply:
Below, why would `task_to_notify` ever be a member of its subscription list if we have:

https://github.com/purduesigbots/pros/blob/f34a219cb49a6b5a8e81b6852abe4c84efa975d0/src/rtos/task_notify_when_deleting.c#L45

and 

https://github.com/purduesigbots/pros/blob/76a22151b7020398b2a36e2e8c52eb73525cf0d0/src/rtos/task_notify_when_deleting.c#L70

Reply:
(discussed offline: this was a misunderstanding stemming from the mistake I made that was called out in the review comment below this one)